ddtrace/_trace/tracer.py

```diff
@@ -298,7 +301,10 @@ def _atexit(self) -> None:
         self.shutdown(timeout=self.SHUTDOWN_TIMEOUT)
 
     def sample(self, span):
-        self._sampler.sample(span)
+        if self._sampler is not None:
+            self._sampler.sample(span)
+        else:
+            log.debug("No sampler available to sample span")
```

Comment:
Should this be a warning? Is there a valid scenario where no samplers are set

Reply:
You're right, warning is probably better.